runtime/shared_common/shrinit.cpp

```diff
@@ -5344,4 +5354,16 @@ findExistingCacheLayerNumbers(J9JavaVM* vm, const char* ctrlDirName, const char*
 	return;
 }
 
+/**
+ * Disables reads/writes to the shared class cache and marks the shared class cache as disabled during a CRIU restore
+ *
+ * @param [in] vm  The Java VM
+**/
+void
+j9shr_disableSharedClassCacheForCriuRestore(J9JavaVM* vm)
+{
+	vm->sharedClassConfig->runtimeFlags |= J9SHR_RUNTIMEFLAG_DENY_CACHE_ACCESS | J9SHR_RUNTIMEFLAG_DENY_CACHE_UPDATES;
+	vm->sharedCacheAPI->xShareClassCacheDisabledOnCRIURestore = TRUE;
```

Comment:
@dsouzai JIT can check this variable `vm->sharedCacheAPI->xShareClassCacheDisabledOnCRIURestore`. It is `TRUE` if `vm->sharedClassConfig` is not NULL and "-Xshareclasses:disableOnRestore" presents. (If `vm->sharedClassConfig` is NULL, SCC is already disabled)